pupil-spa/src/app/login/login.component.spec.ts

```diff
@@ -71,7 +121,121 @@ describe('LoginComponent', () => {
     expect(compiled.querySelector('#schoolPin')).toBeTruthy();
     expect(compiled.querySelector('#pupilPin')).toBeTruthy();
   });
+
+  describe('before login submission', () => {
+    it('should set loginPending to false', async () => {
+      expect(component.loginPending).toBeFalsy();
+    });
+  });
+
+  describe('during login submission', () => {
+    it('should set the loginPending to true', async () => {
+      component.onSubmit('goodPin', 'goodPin');
+      expect(component.loginPending).toBeTruthy();
+    });
+  });
+
+  describe('on successful login', () => {
+    beforeEach(() => {
+      promiseHelper.resolve({ success: 'login okay' });
+    });
+
+    it('should set the loginPending to false', async () => {
+      await component.onSubmit('goodPin', 'goodPin');
+      expect(component.loginPending).toBeFalsy();
+    });
+
+    it('should initialise the QuestionService and WarmupQuestionService on login', async () => {
+      component.onSubmit('goodPin', 'goodPin');
+      fixture.whenStable().then(() => {
+        expect(mockRouter.navigate).toHaveBeenCalled();
+        expect(mockQuestionService.initialise).toHaveBeenCalledTimes(1);
+        expect(mockWarmupQuestionService.initialise).toHaveBeenCalledTimes(1);
+        expect(mockRegisterInputService.initialise).toHaveBeenCalledTimes(1);
+        expect(mockPupilPrefsService.loadPupilPrefs).toHaveBeenCalled();
+      });
+    });
+
+    it('should reject a second submit', async () => {
```

Comment:
🥇 maybe change the wording to 'prevent' rather than reject, as reject suggests the user would have been rejected in some way

Reply:
changed